lib/counter_culture.rb

```diff
@@ -286,7 +286,11 @@ def change_counter_cache(options)
       if id_to_change && options[:counter_column]
         delta_magnitude = if options[:delta_column]
                             delta_attr_name = options[:was] ? "#{options[:delta_column]}_was" : options[:delta_column]
-                            self.send(delta_attr_name) || 0
+                            val = self.send(delta_attr_name)
+                            # check if value is a string. This can happen if the
+                            # delta_column is an enum
+                            val = val.is_a?(String) ? self.class.send(options[:delta_column].pluralize)[val] : val
```

Comment:
I'm not a big fan of checking for `String` here. Is there a way to more explicitly check whether the column is an enum and then explicitly unbox the value? This seems a little weird and brittle to me, in all honesty. You had something in the issue you had opened before that seems closer to something I'd want to do here.